File: win-prog/ProgWinVB/Images and Bitmaps/ImageAtPoints/ImageAtPoints.vb
'----------------------------------------------
' ImageAtPoints.vb (c) 2002 by Charles Petzold
'----------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class ImageAtPoints
    Inherits PrintableForm

    Private img As image

    Shared Shadows Sub Main()
        Application.Run(new ImageAtPoints())
    End Sub

    Sub New()
        Text = "Image At Points"
        img = image.FromFile("..\..\Apollo11FullColor.jpg")
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        grfx.DrawImage(img, New Point() {New Point(cx \ 2, 0), _
                                         New Point(cx, cy \ 2), _
                                         New Point(0, cy \ 2)})
    End Sub
End Class

File: win-prog/ProgWinCS/Images and Bitmaps/ImageAtPoints/ImageAtPoints.cs
//--------------------------------------------
// ImageAtPoints.cs © 2001 by Charles Petzold
//--------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class ImageAtPoints: PrintableForm
{
     Image image;

     public new static void Main()
     {
          Application.Run(new ImageAtPoints());
     }
     public ImageAtPoints()
     {
          Text = "Image At Points";

          image = Image.FromFile("..\\..\\..\\Apollo11FullColor.jpg");
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          grfx.DrawImage(image, new Point[] { new Point(cx / 2, 0),
                                              new Point(cx,     cy / 2),
                                              new Point(0,      cy / 2)});
               
        
     }
}
